### PR TITLE
Fwxv 211 write uv cutoff smoke tests

### PR DESCRIPTION
**Pull request by Taksh-041102**

### CODE REVIEW

File: projects/fw103_taksh/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<!--
```
Comment (JarvisWeng) [inline projects/fw103_taksh/README.md:1]: Remove all your code not related to this ticket

File: projects/uv_cutoff/inc/uv_cutoff.h [hunk 2]
```diff
@@ -4,3 +4,5 @@ typedef enum UVCutoffState {
   UV_CUTOFF_ACTIVE = 0,
   UV_CUTOFF_DISCONNECTED,
 } UVCutoffState;
+
+void uv_ligc() {}
```
Comment (JarvisWeng) [inline projects/uv_cutoff/inc/uv_cutoff.h:8]: What is this for?

File: smoke/uv_cutoff/src/main.c [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  set_uv_cutoff_notification_signal1(state);
+
+  if (state == UV_CUTOFF_DISCONNECTED) {
+    return;
+  }
+  
+
+  if (get_horn_and_lights_horn_state() && !lights_check) {
+    status = gpio_set_state(&horn, GPIO_STATE_HIGH);
+    // if(status != STATUS_CODE_OK) {
```
Comment (JarvisWeng) [inline smoke/uv_cutoff/src/main.c:61]: You can keep the debug messages or remove the comments

**Review by JarvisWeng — COMMENTED** [review]
Looks mostly good

File: projects/fw103_taksh/README.md (hunk 1 above)
Comment (Taksh-041102) [reply projects/fw103_taksh/README.md:1]: Sorry, I didn't understand what you meant here.

File: projects/uv_cutoff/inc/uv_cutoff.h (hunk 2 above)
Comment (Taksh-041102) [reply projects/uv_cutoff/inc/uv_cutoff.h:8]: I removed a backspace after uv_ligc to resolve an error which scons lint threw.

File: projects/fw103_taksh/README.md (hunk 1 above)
Comment (JarvisWeng) [reply projects/fw103_taksh/README.md:1]: Like `fw103_taksh`, `hello_world` all these files you included shouldn't be included in your PR.
You need to remove them

File: projects/uv_cutoff/inc/uv_cutoff.h (hunk 2 above)
Comment (JarvisWeng) [reply projects/uv_cutoff/inc/uv_cutoff.h:8]: Sorry what was the error? I don't think `uv_ligc` is used anywhere

File: smoke/uv_cutoff/src/main.c [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+#include "log.h"
+#include "misc.h"
+#include "soft_timer.h"
+#include "tasks.h"
+#include "uv_cutoff.h"
+#include "uv_cutoff_getters.h"
+#include "uv_cutoff_setters.h"
+
+// TODO(devAdhiraj): update with actual ports and pins
+#define UV_CUTOFF_PORT GPIO_PORT_A
```
Comment (JarvisWeng) [inline smoke/uv_cutoff/src/main.c:18]: You should look at the UV cutoff schematic to find out these pins. If you can't, then just put random ones for now to get your test passing though.

File: smoke/uv_cutoff/src/main.c (hunk 3 above)
Comment (JarvisWeng) [reply smoke/uv_cutoff/src/main.c:61]: What that means is remove all commented code. If you want to keep them, then uncomment them

File: projects/fw103_taksh/README.md (hunk 1 above)
Comment (Taksh-041102) [reply projects/fw103_taksh/README.md:1]: ah alright, noted!

File: smoke/uv_cutoff/src/main.c [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+  }
+}
+
+TASK(smoke_task, TASK_MIN_STACK_SIZE) {
+  //
+  //TEST 1 - Disconnected UV status
+  gpio_set_state(&uv_status,GPIO_STATE_LOW);
+  uv_smoke_logic();
+  assert( g_tx_struct.uv_cutoff_notification_signal1 == UV_CUTOFF_DISCONNECTED );
+  LOG_DEBUG("uv_cutoff notification signal set to 'UV_CUTOFF_DISCONNECTED'\n");
```
[the remark this answers is not in the document]
Comment (Taksh-041102) [reply smoke/uv_cutoff/src/main.c:124]: yup, made this change. Yet to push it.

File: smoke/uv_cutoff/src/main.c [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+      //   LOG_DEBUG("Lights State set correctly\n");
+      // }
+    }
+  }
+}
+
+TASK(smoke_task, TASK_MIN_STACK_SIZE) {
+  //
+  //TEST 1 - Disconnected UV status
+  gpio_set_state(&uv_status,GPIO_STATE_LOW);
```
[the remark this answers is not in the document]
Comment (Taksh-041102) [reply smoke/uv_cutoff/src/main.c:121]: Haven't pushed it yet.

File: smoke/uv_cutoff/src/main.c (hunk 3 above)
Comment (Taksh-041102) [reply smoke/uv_cutoff/src/main.c:61]: got it, it'll be in the new push.

File: projects/uv_cutoff/inc/uv_cutoff.h (hunk 2 above)
Comment (Taksh-041102) [reply projects/uv_cutoff/inc/uv_cutoff.h:8]: I changed uv_ligc () {} -> uv_ligc() {}.
scons lint said to remove an extra space after uv_ligc.

Comment (JarvisWeng) [reply projects/uv_cutoff/inc/uv_cutoff.h:8]: This is an entirely new line. It shouldn't be here if you're not using it

File: smoke/uv_cutoff/src/main.c [hunk 7]
```diff
@@ -163,7 +160,7 @@ int main() {
   
   gpio_init();
 
-  gpio_init_pin(&uv_status, GPIO_INPUT_PULL_UP, GPIO_STATE_HIGH);
+  gpio_init_pin(&uv_status, GPIO_OUTPUT_PUSH_PULL, GPIO_STATE_HIGH);
```
Comment (JarvisWeng) [inline smoke/uv_cutoff/src/main.c:163]: Write a comment here saying that is should be `GPIO_INPUT_PULL_UP` , but you're faking that for now.

File: smoke/uv_cutoff/src/main.c [hunk 8]
```diff
@@ -54,107 +54,104 @@ void uv_smoke_logic() {
...
-      // } else if (state_val == GPIO_STATE_HIGH) {
-      //   LOG_DEBUG("Horn State set correctly\n");
-      // }
     }
   } else {
     status = gpio_set_state(&horn, GPIO_STATE_LOW);
-    // if(status != STATUS_CODE_OK) {
-    //   LOG_DEBUG("GPIO Horn State could not be set to low\n");
-    // }
+    g_rx_struct.horn_and_lights_horn_state = GPIO_STATE_LOW;
```
[the remark this answers is not in the document]
Comment (Taksh-041102) [reply smoke/uv_cutoff/src/main.c:70]: okay, I'll make that change.

File: smoke/uv_cutoff/src/main.c [hunk 9]
```diff
@@ -54,107 +54,104 @@ void uv_smoke_logic() {
   if (state == UV_CUTOFF_DISCONNECTED) {
     return;
   }
-  
 
-  if (get_horn_and_lights_horn_state() && !lights_check) {
```
[the remark this answers is not in the document]
Comment (Taksh-041102) [reply smoke/uv_cutoff/src/main.c]: Yup, it is. I see what you mean by treating like an input. noted!

File: smoke/uv_cutoff/src/main.c (hunk 7 above)
Comment (Taksh-041102) [reply smoke/uv_cutoff/src/main.c:163]: fixed :)

File: projects/uv_cutoff/inc/uv_cutoff.h (hunk 2 above)
Comment (JarvisWeng) [reply projects/uv_cutoff/inc/uv_cutoff.h:8]: You should just delete this

Comment (Taksh-041102) [reply projects/uv_cutoff/inc/uv_cutoff.h:8]: I had forgot to make this change, it'll be their in the new PR.

**Pull request merged**